lib/index.d.ts

```diff
@@ -39,3 +45,76 @@ export class StateController<State> {
   /** @internal Unsubscribe the listener callback that was passed to subscribeUpdates */
   unsubscribeUpdates(listener: (props: Partial<State>) => void): void;
 }
+
+export class Component<State> extends WebComponent {
+    /**
+     * Defines the state of the component, including all the properties required for rendering.
+     */
+    state: State;
```

Comment:
I believe @tdumitrescu added ability for child components to have separate `state` and `appState`